server/src/main/java/org/opensearch/cluster/hahealth/ClusterHaHealth.java

```diff
@@ -0,0 +1,358 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.cluster.hahealth;
+
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.cluster.health.ClusterIndexHealth;
+import org.opensearch.cluster.metadata.IndexMetadata;
+import org.opensearch.cluster.metadata.WeightedRoutingMetadata;
+import org.opensearch.cluster.node.DiscoveryNode;
+import org.opensearch.cluster.routing.IndexRoutingTable;
+import org.opensearch.cluster.routing.ShardRouting;
+import org.opensearch.cluster.routing.WeightedRouting;
+import org.opensearch.cluster.routing.allocation.AwarenessReplicaBalance;
+import org.opensearch.cluster.routing.allocation.decider.AwarenessAllocationDecider;
+import org.opensearch.common.collect.ImmutableOpenMap;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.io.stream.Writeable;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Cluster state HA health information
+ *
+ */
+public class ClusterHaHealth implements Writeable {
+
+    private final int numberOfNodes;
+    private final int numberOfDataNodes;
+    private final int activeShards;
+    private final int unassignedShards;
+    private final int initializingShards;
+    private final ClusterHealthStatus status;
+    private final Map<String, Object> collatedAttributeHealth;
+
+    public static final String NAME = "name";
+    public static final String NODES = "nodes";
+    public static final String ACTIVE_SHARDS = "shards";
+    public static final String UNASSIGNED_SHARDS = "unassignedShards";
+    public static final String WEIGHTS = "weights";
+
+    /**
+     * Creates cluster ha_health from cluster state.
+     *
+     * @param clusterState The current cluster state. Must not be null.
+     * @param clusterSettings the current cluster settings.
+     */
+    public ClusterHaHealth(ClusterState clusterState, ClusterSettings clusterSettings) {
```

Comment:
The constructor is just too big, would prefer smaller methods

Reply:
Added smaller methods